allen-cell-animated/cell-feature-explorer
# Fix NaN error

Problem
=======
The plot was giving a lot of weird looking NaN errors. And the checkboxes were inconsistent. 
[Link to story or ticket](https://aicsjira.corp.alleninstitute.org/browse/CFE-61)

Solution
========
1. converted the NaNs to nulls. We could move this change up into the data flow too because JSON can handle nulls. But this check here seemed like a good check incase someone made a mistake formatting their data. 
2. If one of the arrays sent to the plot was all nulls, plotly threw an error. Which is the case for the default plot on the FISH dataset, only one gene pair is represented in that plot. So I just do a check and return an empty array if there are no values. 

## Type of change
Please delete options that are not relevant.

* Bug fix (non-breaking change which fixes an issue)


Steps to Verify:
----------------
1. npm run start:dev-db
2. load the fish dataset, confirm there are no plotly errors
3. Uncheck the orange MYH6-MYH7 (only set being plotted)
4. plot should be empty, without any errors in the console
